provider/azure/azure_discover.go

```diff
@@ -98,26 +99,30 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
 func fetchAddrsWithTags(tagName string, tagValue string, vmnet network.InterfacesClient, l *log.Logger) ([]string, error) {
 	// Get all network interfaces across resource groups
 	// unless there is a compelling reason to restrict
-	netres, err := vmnet.ListAll()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	netres, err := vmnet.ListAll(ctx)
+
 	if err != nil {
 		return nil, fmt.Errorf("discover-azure: %s", err)
 	}
 
-	if netres.Value == nil {
+	if len(netres.Values()) == 0 {
 		return nil, fmt.Errorf("discover-azure: no interfaces")
 	}
 
 	// Choose any PrivateIPAddress with the matching tag
 	var addrs []string
-	for _, v := range *netres.Value {
+	for _, v := range netres.Values() {
```

Comment:
Can you elaborate on why this removed pointer dereferencing but somewhat inconsistently? for example the ID in `v`  is dereferenced using the pointer but not the tags.

Reply:
Value was removed as a field on a PrivateIPAddress and was replaced with `Values()`: https://github.com/Azure/azure-sdk-for-go/blob/master/services/network/mgmt/2015-06-15/network/models.go#L5882

I can go through and audit other areas, although I'm definitely relying quite a lot on the compiler.

Reply:
ok  v.Tags is working because of the syntactic sugar in Go for dereferencing- specifically this _As an exception, if the type of x is a defined pointer type and (*x).f is a valid selector expression denoting a field (but not a method), x.f is shorthand for (*x).f._  from https://golang.org/ref/spec#Selectors. 
For clarity sake, can you make *v.ID be v.ID ? That makes both accessors consistent and helps readability.